Changes

```diff
@@ -258,12 +258,14 @@ OCaml 5.0
   Guillaume Munch-Maccagnoni, additional discussions with Stephen
   Dolan and Luc Maranget)
 
-- #5299, #4787, #11138, #11272: `Caml_state` is `NULL` when the domain lock is
-  not held. This allows to test if the current thread holds the lock of its
-  domain, and it makes it simpler to debug when one forgets to acquire
-  the lock.
-  (Guillaume Munch-Maccagnoni, review by Sadiq Jaffer, Xavier Leroy and
-  Gabriel Scherer)
+- #5299, #4787, #11138, #11272, #?: The thread-local variable
+  `Caml_state_opt` is introduced, which is `NULL` when the domain lock
+  is not held, and holds the pointer to the domain-local state
+  otherwise. This allows to test from C code if the current thread
```

Comment:
The following description would be easier to understand for me.

> To help debugging, `Caml_state` now dynamically checks that the domain lock is held and fails otherwise.
> A new variable `Caml_state_opt` can always be accessed, and is `NULL` when the domain lock is not held.

Reply:
Good suggestion, thanks!